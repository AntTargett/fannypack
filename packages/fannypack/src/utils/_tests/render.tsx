import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
