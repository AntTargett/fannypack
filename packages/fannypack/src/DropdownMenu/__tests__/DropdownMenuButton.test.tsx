import * as React from 'react';
import { DropdownMenu } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Button {...menu}>Actions</DropdownMenu.Button>}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for CSS props', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu.Button {...menu} backgroundColor="red">
            Actions
          </DropdownMenu.Button>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('DropdownMenu.Button.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu.Button
            {...menu}
            overrides={{ DropdownMenu: { Button: { css: { root: { backgroundColor: 'red' } } } } }}
          >
            Actions
          </DropdownMenu.Button>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('DropdownMenu.Button.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Button {...menu}>Actions</DropdownMenu.Button>}
      </DropdownMenu.State>,
      { theme: { DropdownMenu: { Button: { css: { root: { backgroundColor: 'red' } } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Button {...menu}>Actions</DropdownMenu.Button>}
      </DropdownMenu.State>,
      { theme: { DropdownMenu: { Button: { defaultProps: { className: 'test', color: 'primary' } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
