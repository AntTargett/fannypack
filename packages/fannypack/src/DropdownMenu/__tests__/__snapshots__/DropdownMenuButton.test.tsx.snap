// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-1[disabled],
.emotion-1[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-1:hover {
  background-color: #efefef;
}

.emotion-1:hover:active {
  background-color: #d6d6d6;
}

.emotion-3 {
  color: #3722ed !important;
}

<div
  aria-controls="test"
  aria-disabled="false"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 emotion-3 test fp-Box fp-Button fp-DropdownMenuButton"
  orientation="vertical"
  role="button"
  tabindex="0"
  type="button"
>
  Actions
</div>
`;

exports[`overrides DropdownMenu.Button.css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-1[disabled],
.emotion-1[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-1:hover {
  background-color: #efefef;
}

.emotion-1:hover:active {
  background-color: #d6d6d6;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-controls="test"
  aria-disabled="false"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 emotion-3 emotion-4 fp-Box fp-DropdownMenuDisclosure fp-Button fp-DropdownMenuButton"
  orientation="vertical"
  role="button"
  tabindex="0"
  type="button"
>
  Actions
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-1[disabled],
.emotion-1[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-1:hover {
  background-color: #efefef;
}

.emotion-1:hover:active {
  background-color: #d6d6d6;
}

<div
  aria-controls="test"
  aria-disabled="false"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-DropdownMenuDisclosure fp-Button fp-DropdownMenuButton"
  orientation="vertical"
  role="button"
  tabindex="0"
  type="button"
>
  Actions
</div>
`;

exports[`props should render correctly for CSS props 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-1[disabled],
.emotion-1[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-1:hover {
  background-color: #efefef;
}

.emotion-1:hover:active {
  background-color: #d6d6d6;
}

.emotion-3 {
  background-color: red !important;
}

<div
  aria-controls="test"
  aria-disabled="false"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-DropdownMenuDisclosure fp-Button fp-DropdownMenuButton"
  orientation="vertical"
  role="button"
  tabindex="0"
  type="button"
>
  Actions
</div>
`;

exports[`theming DropdownMenu.Button.css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-1[disabled],
.emotion-1[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-1:focus {
  outline: unset;
  z-index: 2;
  box-shadow: gray 0px 0px 0px 1px, #a6a6a6 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-1:hover {
  background-color: #efefef;
}

.emotion-1:hover:active {
  background-color: #d6d6d6;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-controls="test"
  aria-disabled="false"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 emotion-3 emotion-4 fp-Box fp-DropdownMenuDisclosure fp-Button fp-DropdownMenuButton"
  orientation="vertical"
  role="button"
  tabindex="0"
  type="button"
>
  Actions
</div>
`;
